rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ThreadDumpView.java

```diff
@@ -75,144 +72,98 @@ public ThreadDumpView(RMController controller) {
 
-    private void createLayout(HLayout executeAndLoading) {
-        this.layout = new VLayout(7);
-        this.layout.setWidth100();
-        this.layout.setHeight100();
-
-        this.layout.addMember(this.nodeLabel);
-        this.layout.addMember(executeAndLoading);
-        this.layout.addMember(this.threadDumpArea);
+    private void initLabels() {
+        this.nodeLabel = new Label();
+        this.nodeLabel.setHeight(25);
+        this.nodeLabel.setContents(NO_NODE_SELECTED_MESSAGE);
 
-        this.layout.hide();
+        this.loadingLabel = new Label();
+        this.loadingLabel.setHeight(25);
+        this.loadingLabel.setIcon("loading.gif");
+        this.loadingLabel.hide();
     }
 
-    private void createThreadDumpButtons() {
+    private void initThreadDumpButtons() {
         this.nodeThreadDumpButton = new IButton("Fetch Node's Thread Dump");
-        this.nodeThreadDumpButton.setWidth("220px");
+        this.nodeThreadDumpButton.setWidth("160px");
         this.nodeThreadDumpButton.addClickHandler(event -> {
             if (this.nodeUrl != null) {
                 String nodeUrl = this.nodeUrl;
                 this.loadingLabel.show();
                 this.controller.getRMService().getNodeThreadDump(LoginModel.getInstance().getSessionId(),
                                                                  nodeUrl,
-                                                                 new AsyncCallback<String>() {
-                                                                     @Override
-                                                                     public void onFailure(Throwable caught) {
-                                                                         LogModel.getInstance()
-                                                                                 .logCriticalMessage(caught.getMessage());
-                                                                         ThreadDumpView.this.loadingLabel.hide();
-                                                                         ThreadDumpView.this.threadDumpArea.setText("Thread dump could not be fetched. See Help -> Display logs for more details.");
-                                                                     }
-
-                                                                     @Override
-                                                                     public void onSuccess(String result) {
-                                                                         ThreadDumpView.this.loadingLabel.hide();
-                                                                         ThreadDumpView.this.threadDumpArea.setText(result);
-                                                                     }
-                                                                 });
+                                                                 getThreadDumpCallback());
             }
         });
+        this.nodeThreadDumpButton.disable();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "nodeUrl" which hides the field declared at line 62. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)